azure-devops/projects/io-backend-projects/io-openid-provider.tf

```
@@ -0,0 +1,80 @@
+variable "io-openid-provider" {
+  default = {
+    repository = {
+      organization   = "pagopa"
+      name           = "io-openid-provider"
+      branch_name    = "master"
+      pipelines_path = ".devops"
+    }
+    pipeline = {
+      cache_version_id = "v3"
+    }
+  }
+}
+
+#
+# Code Review pipeline
+#
+
+# Define code review pipeline
+resource "azuredevops_build_definition" "io-openid-provider-code-review" {
+  depends_on = [azuredevops_serviceendpoint_github.io-azure-devops-github-pr, azuredevops_project.project]
+
+  project_id = azuredevops_project.project.id
+  name       = "${var.io-openid-provider.repository.name}.code-review"
+  path       = "\\${var.io-openid-provider.repository.name}"
+
+  pull_request_trigger {
+    initial_branch = var.io-openid-provider.repository.branch_name
+    forks {
+      enabled       = false
+      share_secrets = false
+    }
+    override {
+      auto_cancel = false
+      branch_filter {
+        include = [var.io-openid-provider.repository.branch_name]
+      }
+      path_filter {
+        exclude = []
+        include = []
+      }
+    }
+  }
+
+  repository {
+    repo_type             = "GitHub"
+    repo_id               = "${var.io-openid-provider.repository.organization}/${var.io-openid-provider.repository.name}"
+    branch_name           = var.io-openid-provider.repository.branch_name
+    yml_path              = "${var.io-openid-provider.repository.pipelines_path}/code-review-pipelines.yml"
+    service_connection_id = azuredevops_serviceendpoint_github.io-azure-devops-github-pr.id
+  }
+
+  variable {
+    name         = "DANGER_GITHUB_API_TOKEN"
+    secret_value = module.secrets.values["DANGER-GITHUB-API-TOKEN"].value
+    is_secret    = true
+  }
+}
+
+# Allow code review pipeline to access Github readonly service connection, needed to access external templates to be used inside the pipeline
+resource "azuredevops_resource_authorization" "io-openid-provider-code-review-github-ro-auth" {
+  depends_on = [azuredevops_serviceendpoint_github.pagopa, azuredevops_build_definition.io-openid-provider-code-review, azuredevops_project.project]
+
+  project_id    = azuredevops_project.project.id
+  resource_id   = azuredevops_serviceendpoint_github.pagopa.id
+  definition_id = azuredevops_build_definition.io-openid-provider-code-review.id
+  authorized    = true
+  type          = "endpoint"
+}
+
+# Allow code review pipeline to access Github pr service connection, needed to checkout code from the pull request branch
+resource "azuredevops_resource_authorization" "io-openid-provider-code-review-github-pr-auth" {
+  depends_on = [azuredevops_serviceendpoint_github.io-azure-devops-github-pr, azuredevops_build_definition.io-openid-provider-code-review, azuredevops_project.project]
+
+  project_id    = azuredevops_project.project.id
+  resource_id   = azuredevops_serviceendpoint_github.io-azure-devops-github-pr.id
```

```suggestion
  resource_id   = azuredevops_serviceendpoint_github.io-azure-devops-github-rw.id
```
for private repository we need rw permissions

---

Repository is now public :)